internal/deprecation/deprecation_plugin.go

```diff
@@ -0,0 +1,81 @@
+package deprecation
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	datasourceschema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	resourceschema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+type Model struct {
+	IsDeprecated         types.Bool   `tfsdk:"is_deprecated"`
+	DeprecationAnnounced types.String `tfsdk:"deprecation_announced"`
+	UnavailableAfter     types.String `tfsdk:"unavailable_after"`
+}
+
+type DeprecationModel = Model //nolint:revive
+
+func NewModel(_ context.Context, in hcloud.Deprecatable) (Model, diag.Diagnostics) {
```

Comment:
Not sure this works with `type DeprecationModel = Model`, IIUC Go considers these two to be distinct types, so you can not easily assign values of one type to the other.

I might have misunderstood Go though.

Reply:
The type alias is different from creating a new type (`with =`)

https://go.dev/play/p/4fKebHRoX9d

Reply:
TIL!